### PR TITLE
[WIP] B4 & D2 Add metadata header and Tag

### PR DESCRIPTION
**Pull request by petrabarus**
- [ ] Metadata header
- [ ] Snapshot tag

### CODE REVIEW

File: src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotFileSegment.java [hunk 1]
```diff
@@ -118,9 +129,10 @@ public void setData(byte[] data) {
     public static SnapshotFileSegment createFromHeaderBytes(byte[] bytes) {
         SnapshotFileSegment snapshot = new SnapshotFileSegment();
         snapshot.type = getTypeByteBytes(Arrays.copyOfRange(bytes, 0, MAGIC_BYTES_KEY.length));
-        snapshot.timestamp = ByteHelper.byteArrayToLittleEndianLong(Arrays.copyOfRange(bytes, 6, 14));
-        snapshot.size = ByteHelper.byteArrayToLittleEndianLong(Arrays.copyOfRange(bytes, 14, 22));
-        snapshot.checksum = Arrays.copyOfRange(bytes, 22, 42);
+        snapshot.relativeTimestamp = ByteHelper.byteArrayToLittleEndianLong(Arrays.copyOfRange(bytes, 6, 14));
+        snapshot.absoluteTimestamp = ByteHelper.byteArrayToLittleEndianLong(Arrays.copyOfRange(bytes, 14, 22));
```
Comment (julianghionoiu) [inline src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotFileSegment.java:133]: We do not need to store the absoluteTimestamp for each snapshot. It should only be written once at the beginning of the file, before the first snapshot. The beginning of the file will act as a header and will grow to contain more information about the context of the recording.

File: src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotsFileWriter.java [hunk 2]
```diff
@@ -45,14 +45,20 @@ public void takeSnapshot() {
...
+            segment.absoluteTimestamp = getTimestamp();
             segment.setData(snapshot.getData());
             IOUtils.write(segment.asBytes(), outputStream);
         } catch (IOException ex) {
             Logger.getLogger(SnapshotsFileWriter.class.getName()).log(Level.SEVERE, null, ex);
         }
     }
 
+    public int getTimestamp() {
+        Long unixTimestamp = System.currentTimeMillis() / 1000L;
```
Comment (julianghionoiu) [inline src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotsFileWriter.java:58]: The time information should come from a timeSource. I always consider time to be an external dependency.
In this case we do not use this time information for scheduling or controlling threads so it is pure data and this data comes from outside.

I would create a class to represent the Metadata - information about recording and explicitly add it to the constructor:
```
       SourceCodeRecorder sourceCodeRecorder = new SourceCodeRecorder.Builder(..)
                .withMetadata(recordingTimestampMillis)
                .withTimeSource(new FakeTimeSource())
```

Comment (julianghionoiu) [reply src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotsFileWriter.java:58]: When we add the header to the file we should also add a magic number for the file identifier.
I thing it should be SRCSTM

Comment (petrabarus) [reply src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotsFileWriter.java:58]: sorry i misread the description

**Pull request closed without merging**